### PR TITLE
Support global processing

### PR DESCRIPTION
**Pull request by jhkennedy**
This allows global processing with autoRIFT using the new JPL parameter files

It currently downloads all the DEM files for the right region -- when autoRIFT can support `/vsis3/` and hybrid local and remote files, this could be optimized.

### CODE REVIEW

File: hyp3_autorift/geometry.py [hunk 1]
```diff
@@ -89,59 +86,37 @@ def bounding_box(safe, priority='reference', polarization='hh', orbits='Orbits',
...
-        dem_file = f'/vsicurl/http://{ITS_LIVE_BUCKET}.s3.amazonaws.com/{AUTORIFT_PREFIX}/{dem}_h.tif'
-        log.info(f'Checking DEM: {dem_file}')
-        dem_ds = gdal.Open(dem_file, gdal.GA_ReadOnly)
-        dem_sr = dem_ds.GetSpatialRef()
-        log.debug(f'DEM projection: {dem_sr}')
 
-        latlon = osr.SpatialReference()
-        latlon.ImportFromEPSG(4326)
+def find_jpl_dem(lat_limits, lon_limits):
+    shape_file = f'/vsicurl/http://{ITS_LIVE_BUCKET}.s3.amazonaws.com/{AUTORIFT_PREFIX}/autorift_parameters.shp'
```
Comment (asjohnston-asf) [inline hyp3_autorift/geometry.py:102]: should be `https` I think

File: hyp3_autorift/geometry.py [hunk 2]
```diff
@@ -89,59 +86,37 @@ def bounding_box(safe, priority='reference', polarization='hh', orbits='Orbits',
...
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    shapes = driver.Open(shape_file, gdal.GA_ReadOnly)
 
-        trans = osr.CoordinateTransformation(latlon, dem_sr)
+    extent = polygon_from_bbox(lat_limits, lon_limits)
+    for feature in shapes.GetLayer(0):
+        if feature.geometry().Contains(extent.Centroid()):
```
Comment (asjohnston-asf) [inline hyp3_autorift/geometry.py:108]: compute the centroid once prior to the loop

File: hyp3_autorift/geometry.py (hunk 1 above)
Comment (asjohnston-asf) [reply hyp3_autorift/geometry.py:102]: disregard, needs to be `http`

File: tests/test_geometry.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    lon_limits = (-40, -41)
+    assert geometry.find_jpl_dem(lat_limits, lon_limits) == 'S24_0240m'
+
+    lat_limits = (-55, -56)
+    lon_limits = (40, 41)
+    assert geometry.find_jpl_dem(lat_limits, lon_limits) == 'S37_0240m'
+
+    lat_limits = (-56, -57)
+    lon_limits = (40, 41)
+    assert geometry.find_jpl_dem(lat_limits, lon_limits) == 'SPS_0240m'
```
Comment (asjohnston-asf) [inline tests/test_geometry.py:34]: should add a `pytest.raises(DemError)` test case

**Comment by asjohnston-asf** [comment]
What's the motivation for moving the isce imports into the individual functions?

**Comment by jhkennedy** [comment]
Mostly to remove a lot of the ISCE chatter from the logs if I wasn't actually using ISCE -- it always gives an annoyingly verbose notice when it's first imported

File: hyp3_autorift/geometry.py (hunk 2 above)
Comment (jhkennedy) [reply hyp3_autorift/geometry.py:108]: good point! (see what I did there?)

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**